src/libra_py/CP2K_methods.py

```diff
@@ -161,31 +161,33 @@ def cube_file_names_cp2k( static_or_dynamics, project_name, min_band, max_band,
 
 
 
-def read_cp2k_tddfpt_log_file( tddfpt_log_file_name, num_ci_states, tolerance ):
+def read_cp2k_tddfpt_log_file( params ):
     """
     This function reads log files generated from TD-DFPT calculations using CP2K and returns the TD-DFPT
     excitation energies, Slater determinent states in terms of the Kohn-Sham orbital indicies, 
     and the Slater determinent coefficients that comprise the multi-configurational electronic states
     
     Args:
-
-        tddfpt_log_file_name ( string ): name of the log file for this particular timestep
-
-        num_ci_states ( int ): how many ci states to consider
-
-        tolerance ( float ): the tolerance for accepting SDs are members of the CI wavefunctions
-
+      params ( dictionary ): parameters dictionary 
+          logfile_name ( string ): name of the log file for this particular timestep
+          num_ci_states ( int ): how many ci states to consider
+          tolerance ( float ): the tolerance for accepting SDs are members of the CI wavefunctions
+          isUSK ( Boolean ): flag for whether or not a spin-polarized Kohn-Sham basis is being used. TRUE means
+                           that a spin-polarized Kohn-Sham basis is being used.
     Returns:
-
        excitation_energies ( list ): The excitation energies in the log file.
-
        ci_basis ( list ): The CI-basis configuration.
-
        ci_coefficients ( list ): The coefficients of the CI-states.
-
+       spin_components (list): The spin component of the excited states.
     """
-    
-    f = open(tddfpt_log_file_name,'r')
+
```

Comment:
please add the input correctness check with:

    comn.check_input(params, default_params, critical_params)

Please see dynamics/tsh/compute/run_dynamics for instance